diff --git a/os/src/task/id.rs b/os/src/task/id.rs
--- a/os/src/task/id.rs
+++ b/os/src/task/id.rs
@@ -53,4 +53,5 @@
 impl Drop for PidHandle {
+    #[inline(never)]
     fn drop(&mut self) {
         //println!("drop pid {}", self.0);
         PID_ALLOCATOR.exclusive_access().dealloc(self.0);
@@ -87,3 +88,4 @@
 impl Drop for KernelStack {
+    #[inline(never)]
     fn drop(&mut self) {
         let (kernel_stack_bottom, _) = kernel_stack_position(self.0);
